### PR TITLE
Update to use cmake presets

### PR DESCRIPTION
**Pull request by adam4813**
Change to use CMake presets and update documentation generation, and the README, along side various bug fixes.

## Description
Update README and added docs generation targets.

Fix readthedoc setup by including python requirements.

Update vcpkg commit hash for dependencies.

Change to use CMake presets with ninja multi config.

Fix asset failing to load missing null pointer check for animation.cpp.

Some adjustments to fmt.

Fix Dead ImGUI root window id being empty.

Add EOL find utility for md5anim and obj parsing. This should be replaced by a better check @elantzb .

Add better validation to md5anim parsing.

Remove explicit default constructor for Mesh.hpp Vertex

Update CMake min version to 19.

Fix include in pxiel-buffer.cpp being nested under tec namespace.
The compiler was errantly looking for tec::std::man because of this.

Add launch config for vscode.

Properly set c++17 via target_compile_features interface to propagate

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by adam4813** [comment]
@elantzb I cannot mark you as a reviewer, but tagging you here

**Review by elantzb — COMMENTED** [review]
Nothing jumps out at me as problematic. I want to try building this and see what happens on my end.

**Comment by elantzb** [comment]
Update this to use latest c++ language standard.

**Comment by elantzb** [comment]
I recommend updating the description on the PR/commit to be more verbose. For the sake of a paper trail if needed in the future.

**Comment by adam4813** [comment]
This will require https://github.com/protocolbuffers/protobuf/pull/10232 to be merged into a new version and the the version needs to be put on VCPKG and the baseline commit hash needs to be updated.

File: README.md [hunk 1]
```diff
@@ -8,74 +8,116 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
+3. `VCPKG_ROOT` must be set a an environment variable, to wherever VCPKG was cloned to
+4. Run `$(VCPKG_ROOT)/bootstrap-vcpkg.[bat|sh]` to setup vcpkg. Use the correct extension for your OS.
+   1. [OPTIONAL] `$(VCPKG_ROOT)/./vcpkg integrate install`
+5. Open the folder in an editor OR see step 5
+   1. [Visual Studio](https://docs.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170)
+   2. [CLion](https://www.jetbrains.com/help/clion/cmake-presets.html)
+6. Run CMake
+   1. CLI
+      1. `cmake --preset=ninja-multi-vcpkg`
+      2. `cmake --build --preset=BUILD_PRESERT` see [presets](#presets)
```
Comment (elantzb) [inline README.md:40]: PRESERT?

File: README.md [hunk 2]
```diff
@@ -8,74 +8,116 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
-   1. `cmake -DCMAKE_TOOLCHAIN_FILE=**VCPKG_DIR**/scripts/buildsystems/vcpkg.cmake ..` in the build directory
-   1. `make` in the build directory
-
-### Part 3 (Unit Tests)
-To generate the unit tests, follow the same instructions from before, but set to true the flag `BUILD_TESTS_TEC`
-
-## Clang Format
+TEC requires cmake 3.19, a c++17 compatible compiler, and a few libraries GLFW3, GLM, ASIO, Protobuf, GLEW, Lua, Bullet, Dear ImGui, sol3, Spdlog and OpenAL.
+
+The libraries are automatically installed when using CMake and VCPKG manifest mode.
```
Comment (elantzb) [inline README.md:13]: Manifest mode in VS may require going to Project Properties > vcpkg and enabling it.

Comment (adam4813) [reply README.md:13]: I cannot find a setting around this, and manifest mode should be solely in the realm of VCPKG https://vcpkg.readthedocs.io/en/latest/users/manifests/

File: README.md (hunk 1 above)
Comment (adam4813) [reply README.md]: Fixed

**Comment by adam4813** [comment]
Visual studio 2017 is not supported any longer it appears due to sol2 https://github.com/ThePhD/sol2/issues/1362

File: README.md [hunk 3]
```diff
@@ -8,74 +8,113 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
+
+`build-essentials pkg-config tar curl zip unzip gdb libgl1-mesa-dev xorg-dev libglu1-mesa-dev libxinerama-dev libxcursor-dev`
+
+## Build Steps
+1. Install [CMake](https://cmake.org/download/)
+2. Install [VCPK](git clone https://github.com/Microsoft/vcpkg.git) - `git clone https://github.com/Microsoft/vcpkg.git`
+3. `VCPKG_ROOT` must be set a an environment variable, to wherever VCPKG was cloned to
+4. Run `$(VCPKG_ROOT)/bootstrap-vcpkg.[bat|sh]` to setup vcpkg. Use the correct extension for your OS.
+   1. [OPTIONAL] `$(VCPKG_ROOT)/./vcpkg integrate install`
+5. Open the folder in an editor OR see step 5
```
Comment (elantzb) [inline README.md:34]: "or see step 5" This *is* step 5.

File: README.md [hunk 4]
```diff
@@ -8,74 +8,113 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
+   1. [Visual Studio](https://docs.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170)
+   2. [CLion](https://www.jetbrains.com/help/clion/cmake-presets.html)
+6. Run CMake
+   1. CLI
+      1. `cmake --preset=ninja-multi-vcpkg`
+      2. `cmake --build --preset=BUILD_PRESET` see [presets](#presets)
+   2. GUI - This will only configure the project. CLI steps or Open in an editor are more useful.
+      1. Set the source folder
```
Comment (elantzb) [inline README.md:42]: clarify that source folder is the root of the repo.

File: README.md [hunk 5]
```diff
@@ -8,74 +8,113 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
+      2. Run Configure
```
Comment (elantzb) [inline README.md:43]: Before this, make sure proper compiler selected.

File: README.md [hunk 6]
```diff
@@ -8,74 +8,113 @@ Head onto our [Discord](https://discord.gg/HM8hhbGSjd) for extended support on b
...
+         1. Pick the Ninja Multi-config generator
+         2. Select Specify toolchain file for cross compiling
+         3. The file input should be filled in, but if not point it to `$(VCPKG_ROOT)/scripts/buildsystems/vcpkg.cmake`
+      3. Run Generate
```
Comment (elantzb) [inline README.md:47]: If cmake-gui was used, then here we need to specify opening the SLN file in the C++ IDE and building the solution.

File: README.md (hunk 5 above)
Comment (elantzb) [reply README.md:43]: Hold up, I realize picking "Ninja" is this step. Testing that now...

File: README.md (hunk 4 above)
Comment (elantzb) [reply README.md:42]: Also, specify binary output file?

File: README.md (hunk 3 above)
Comment (adam4813) [reply README.md:34]: I reordered this, I'll update.

**Comment by adam4813** [comment]
Prior run was successful and the final commit should have skipped run as it was a typo fix to the README. I am going to merge this via admin powers instead of waiting.

**Pull request merged**